target_chains/ethereum/contracts/forge-test/PulseScheduler.t.sol

```diff
@@ -542,6 +542,145 @@ contract SchedulerTest is Test, SchedulerEvents, PulseSchedulerTestUtils {
         );
     }
 
+    function testAddFundsEnforcesMinimumBalance() public {
+        // First add a subscription with minimum balance
+        uint256 subscriptionId = addTestSubscription(
+            scheduler,
+            address(reader)
+        );
+
+        // Get subscription parameters and initial balance
+        (
+            SchedulerState.SubscriptionParams memory params,
+            SchedulerState.SubscriptionStatus memory status
+        ) = scheduler.getSubscription(subscriptionId);
+
+        // Calculate minimum balance
+        uint256 minimumBalance = scheduler.getMinimumBalance(
+            uint8(params.priceIds.length)
+        );
+
+        // Verify initial balance is at minimum
+        assertEq(
+            status.balanceInWei,
+            minimumBalance,
+            "Initial balance should equal minimum balance"
+        );
+
+        // Add some funds to the subscription
+        uint256 additionalFunds = 0.1 ether;
+        scheduler.addFunds{value: additionalFunds}(subscriptionId);
+
+        // Verify funds were added
+        (, SchedulerState.SubscriptionStatus memory statusAfterAdd) = scheduler
+            .getSubscription(subscriptionId);
+        assertEq(
+            statusAfterAdd.balanceInWei,
+            minimumBalance + additionalFunds,
+            "Balance should be increased by the added funds"
+        );
+
+        // Now create a new subscription but don't fund it fully
+        SchedulerState.SubscriptionParams
+            memory newParams = createDefaultSubscriptionParams(
+                2,
+                address(reader)
+            );
+
+        // Calculate minimum balance for this new subscription
+        uint256 newMinimumBalance = scheduler.getMinimumBalance(
+            uint8(newParams.priceIds.length)
+        );
+
+        // Try to create with insufficient funds
+        uint256 insufficientFunds = newMinimumBalance - 1 wei;
+        vm.expectRevert(abi.encodeWithSelector(InsufficientBalance.selector));
+        scheduler.createSubscription{value: insufficientFunds}(newParams);
+
+        // Create with sufficient funds
+        uint256 newSubscriptionId = scheduler.createSubscription{
+            value: newMinimumBalance
+        }(newParams);
+
+        // Verify subscription was created with minimum balance
+        (, SchedulerState.SubscriptionStatus memory newStatus) = scheduler
+            .getSubscription(newSubscriptionId);
+        assertEq(
+            newStatus.balanceInWei,
+            newMinimumBalance,
+            "New subscription balance should equal minimum balance"
+        );
```

Comment:
this is not testing balance check in `addFunds`. here you need to send some updates to drain the balance and then try to add funds